.ci/end2end.groovy

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env groovy
+
+library identifier: 'apm@current',
+retriever: modernSCM(
+  [$class: 'GitSCMSource',
+  credentialsId: 'f94e9298-83ae-417e-ba91-85c279771570',
+  id: '37cf2c00-2cc7-482e-8c62-7bbffef475e2',
+  remote: 'git@github.com:elastic/apm-pipeline-library.git'])
+
+pipeline {
+  agent { label 'linux && immutable' }
+  environment {
+    BASE_DIR = 'src/github.com/elastic/kibana'
+    HOME = "${env.WORKSPACE}"
+    APM_ITS = 'apm-integration-testing'
+    CYPRESS_DIR = 'x-pack/legacy/plugins/apm/cypress'
+  }
+  options {
+    timeout(time: 1, unit: 'HOURS')
+    buildDiscarder(logRotator(numToKeepStr: '40', artifactNumToKeepStr: '20', daysToKeepStr: '30'))
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+  }
+  triggers {
+    issueCommentTrigger('(?i).*jenkins\\W+run\\W+(?:the\\W+)?e2e(?:\\W+please)?.*')
+  }
+  parameters {
+    booleanParam(name: 'FORCE', defaultValue: false, description: 'Whether to force the run.')
+  }
+  stages {
+    stage('Checkout') {
+      options { skipDefaultCheckout() }
+      steps {
+        deleteDir()
+        gitCheckout(basedir: "${BASE_DIR}", githubNotifyFirstTimeContributor: false)
+        script {
+          dir("${BASE_DIR}"){
+            def regexps =[ "^x-pack/legacy/plugins/apm/" ]
+            env.APM_UPDATED = isGitRegionMatch(patterns: regexps)
+          }
+        }
+        dir("${APM_ITS}"){
+          git changelog: false,
+              credentialsId: 'f6c7695a-671e-4f4f-a331-acdce44ff9ba',
+              poll: false,
+              url: "git@github.com:elastic/${APM_ITS}.git"
+        }
+      }
+    }
+    stage('Start services') {
+      options { skipDefaultCheckout() }
+      when {
+        anyOf {
+          expression { return params.FORCE }
+          expression { return env.APM_UPDATED != "false" }
+        }
+      }
+      steps {
+        dir("${APM_ITS}"){
+          sh './scripts/compose.py start master --no-kibana --no-xpack-secure'
```

Comment:
I think we want to enable security at some point.

Reply:
That's a good point, let's keep this requirement as a follow-up then :)